lib/java/src/org/apache/thrift/async/TAsyncMethodCall.java

```diff
@@ -225,8 +231,13 @@ private void cleanUpAndFireCallback(SelectionKey key) {
     key.interestOps(0);
     // this ensures that the TAsyncMethod instance doesn't hang around
     key.attach(null);
```

Comment:
Another implication of the current (pre-existing) design is that the client cannot handle unordered or even queued calls.
IMO it should but that would need major rewrite.
(To be clear, testParallelCalls has bunches of concurrent clients each of which does sequential calls)